test/StreamTests.hs

```diff
@@ -0,0 +1,177 @@
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE QuasiQuotes       #-}
+{-# LANGUAGE TupleSections     #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE TypeApplications #-}
+{-# LANGUAGE CPP #-}
+
+module StreamTests
+  ( tests
+  ) where
+
+#ifdef USE_MICROLENS
```

Comment:
I stubbed out the tests module for microlens because it doesn't understand setOf, at that point I had quite enough of microlens.